etcdmain/config.go

```diff
@@ -317,6 +356,141 @@ func (cfg *config) Parse(arguments []string) error {
 		}
 	}
 
+	return nil
+}
+
+func (cfg *config) configFromFile() error {
+	// Read in the contents of the config file.
+	b, err := ioutil.ReadFile(cfg.configFile)
+	if err != nil {
+		return err
+	}
+
+	// Unmarshal the YAML.
+	err = yaml.Unmarshal(b, cfg)
+	if err != nil {
+		return err
+	}
+
+	// Make sure the defaults are set for the missing fields.
+	cfg.setDefaultsIfNeeded()
+
+	// Validate whether conflicting parameters are specified.
+	nSet := 0
```

Comment:
Personally, I feel we should not set any default value for configFile case actually. If we want to provide default setting, we should provide them in the sample configuration file. Missing value will just be missing value. It is on users hand.